x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/AbstractArrayState.java

```diff
@@ -21,7 +21,7 @@ public AbstractArrayState(BigArrays bigArrays) {
         this.bigArrays = bigArrays;
     }
 
-    final boolean hasValue(int groupId) {
+    boolean hasValue(int groupId) {
```

Comment:
I think we wouldn't want to extend this and get `seen` if we're overriding this method.

Reply:
Oh God. Fixed! Added instead a single boolean state, just to know whether using a vector or a block in toBlock()